pkg/handlers/kafka/kafka.go

```diff
@@ -155,6 +142,36 @@ func (h KafkaHandler) Apply(application *aiven_nais_io_v1.AivenApplication, _ *a
 	return nil
 }
 
+func (h KafkaHandler) provideServiceUser(application *aiven_nais_io_v1.AivenApplication, projectName string, serviceName string, secret *v1.Secret, logger *log.Entry) (*aiven.ServiceUser, error) {
+	var aivenUser *aiven.ServiceUser
+	var err error
+	if serviceUserName, ok := secret.GetAnnotations()[ServiceUserAnnotation]; ok {
+		aivenUser, err = h.serviceuser.Get(serviceUserName, projectName, serviceName, logger)
```

Comment:
In a future PR, I think this part of the code should check if the error was NotFound, and in that case fall back to creating the user. This would add a method to recover when someone/something deletes a service user that is still in use. Not very likely to happen, so not doing it in this PR which is already slightly too big.